api/src/main/java/org/openmrs/module/pihcore/htmlformentry/action/RegisterBabyAction.java

```diff
@@ -0,0 +1,192 @@
+            Concept noConcept = Context.getConceptService().getConceptByMapping(NO_CONCEPT, "CIEL");
+
+            Patient patient = formEntrySession.getPatient();
+            Encounter encounter = formEntrySession.getEncounter();
+            boolean updateEncounter = false;
+            for (Obs candidate : encounter.getObsAtTopLevel(false)) {
+                if (candidate.getConcept().equals(newbornDetailsConcept)) {
+                    Set<Obs> groupMembers = candidate.getGroupMembers();
+                    if ( groupMembers != null ) {
+                        for (Obs groupMember : groupMembers) {
+                            // search for Labour and Delivery obs that have the "Register patient in EMR" obs set to No (SL-617)
+                            if (groupMember.getConcept().equals(registerBabyConcept) && groupMember.getValueCoded().equals(noConcept)) {
+                                //find the gender and the birthdate time of the baby
+                                String gender = getObsValue(groupMembers, BABY_GENDER_CONCEPT_SOURCE, BABY_GENDER_CONCEPT);
+                                Date birthDatetime = getObsDateValue(groupMembers, DATE_TIME_OF_BIRTH_CONCEPT_SOURCE, DATE_TIME_OF_BIRTH_CONCEPT);
+                                if (StringUtils.isNotBlank(gender) && (birthDatetime != null)) {
+                                    Patient baby = registerBaby(patient, gender, birthDatetime, encounter.getLocation());
+                                    if (baby != null && baby.getUuid() != null) {
+                                        groupMember.setValueCoded(yesConcept);
+                                        groupMember.setComment(baby.getUuid());
+                                        updateEncounter = true;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            if (updateEncounter) {
+                Context.getEncounterService().saveEncounter(encounter);
+            }
+
+            //automatically redirect to the children page
+            formEntrySession.setAfterSaveUrlTemplate("pihcore/children/children.page?patientId={{patient.id}}");
+        }
+    }
+
+    Patient registerBaby(Patient mother, String gender, Date birthDatetime, Location location) {
+        PersonService personService = Context.getPersonService();
+        RelationshipType motherToChildRelationshipType = personService.getRelationshipTypeByUuid(PihEmrConfigConstants.RELATIONSHIPTYPE_MOTHERTOCHILD_UUID);
+        RelationshipType relationship = personService.getRelationshipTypeByUuid(motherToChildRelationshipType.getUuid());
+        List<Relationship> relationships = new ArrayList<Relationship>();
+        if (relationship !=null) {
+            relationships.add(new Relationship(mother, null, motherToChildRelationshipType));
```

Comment:
I don't understand this.  We are creating a relationship before creating the baby?  And leaving the personB null?  That doesn't seem right.

Reply:
@mseaton , i just follow the same logic/workflow from registrationapp:
https://github.com/openmrs/openmrs-module-registrationapp/blob/master/omod/src/main/java/org/openmrs/module/registrationapp/fragment/controller/RegisterPatientFragmentController.java#L168

That could be incorrect too, but I don't think any relationships are actually saved until the patient is saved?